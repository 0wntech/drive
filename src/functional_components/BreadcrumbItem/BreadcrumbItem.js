import React from 'react';
import styles from './BreadcrumbItem.module.css';
import PropTypes from 'prop-types';

function BreadcrumbItem({ onClick, label, seperator }) {
    return (
        <div onClick={onClick} className={styles.container}>
            <div className={styles.breadcrumb}>{label}</div>
            {seperator ? <div className={styles.seperator}>/</div> : null}
        </div>
    );
}

BreadcrumbItem.propTypes = {
    onClick: PropTypes.func,
    label: PropTypes.string,
    seperator: PropTypes.bool,
};

export default BreadcrumbItem;
