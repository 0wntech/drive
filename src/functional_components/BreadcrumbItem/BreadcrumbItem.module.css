.breadcrumb {
    color: #5c94ff;
    cursor: pointer;
    font-style: normal;
    font-weight: 300;
    font-size: 18px;
}

.seperator {
    margin: 0 10px 0 10px;
    font-style: normal;
    font-weight: 300;
    font-size: 18px;
}

.container {
    display: flex;
}
