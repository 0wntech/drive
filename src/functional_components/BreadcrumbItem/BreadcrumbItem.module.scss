@import '../../styles/constants.scss';

.breadcrumb {
    color: $light-blue;
    cursor: pointer;
    font-style: normal;
    font: $medium-text;
}

.seperator {
    margin: 0 10px 0 10px;
    font-style: normal;
    font: $medium-text;
}

.container {
    display: flex;
}
