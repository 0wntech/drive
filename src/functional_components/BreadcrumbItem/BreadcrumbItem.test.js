import React from 'react';
import BreadcrumbItem from './BreadcrumbItem';
import { shallow } from 'enzyme';
import TestRenderer from 'react-test-renderer';

describe('BreadcrumbItem', () => {
    it('should render correctly', () => {
        const tree = TestRenderer.create(<BreadcrumbItem label="Test" />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should handle the click event', () => {
        const mockFunction = jest.fn();
        const wrapper = shallow(
            <BreadcrumbItem label="Test" onClick={mockFunction} />
        );
        wrapper.simulate('click');
        expect(mockFunction).toHaveBeenCalled();
    });

    xit('renders the label', () => {
        const wrapper = shallow(
            <BreadcrumbItem label="Test" onClick={mockFunction} />
        );
        expect(wrapper.find('.breadcrumb'));
    });
});
