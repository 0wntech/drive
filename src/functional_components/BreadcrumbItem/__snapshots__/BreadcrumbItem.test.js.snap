// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbItem should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="breadcrumb"
    data-test-id="breadcrumb-Test"
  >
    Test
  </div>
</div>
`;
